src/Workspaces/Core/Portable/NamingStyles/EditorConfig/EditorConfigNamingStyleParser.NamingRule.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles
+{
+    internal static partial class EditorConfigNamingStyleParser
+    {
+        private static bool TryGetSerializableNamingRule(
+            string namingRuleTitle,
+            SymbolSpecification symbolSpec,
+            NamingStyle namingStyle,
+            IReadOnlyDictionary<string, object> conventionsDictionary,
+            out SerializableNamingRule serializableNamingRule)
+        {
+            var severity= GetRuleSeverity(namingRuleTitle, conventionsDictionary);
+            serializableNamingRule = new SerializableNamingRule()
+            {
+                EnforcementLevel = severity,
+                NamingStyleID = namingStyle.ID,
+                SymbolSpecificationID = symbolSpec.ID
+            };
+            return true;
+        }
+
+        private static DiagnosticSeverity GetRuleSeverity(
+            string namingRuleName,
+            IReadOnlyDictionary<string, object> conventionsDictionary)
+        {
+            if (conventionsDictionary.TryGetValue($"dotnet_naming_rule.{namingRuleName}.severity", out object result))
+            {
+                return ParseEnforcementLevel(result as string ?? string.Empty);
+            }
+
+            return default(DiagnosticSeverity); ;
+        }
+
+        private static DiagnosticSeverity ParseEnforcementLevel(string ruleSeverity)
+        {
+            switch (ruleSeverity)
+            {
+                case "silent": return DiagnosticSeverity.Hidden;
```

Comment:
Are these severity names reused for all style options? If so, can we put the magic strings in one place and reuse it here?